bindings/rust/src/lib.rs

```diff
@@ -269,6 +380,113 @@ mod tests {
         assert_eq!(v.as_f64(), f64::MAX);
     }
 
+    #[test]
+    fn typed_value_conversion() {
```

Comment:
You need tests for wrong type.

Reply:
Wrong type is a compilation error.

Reply:
I mean the case when `.as_u32()` is `None`.

Reply:
I pushed that 10 minutes ago, just github is slow today showing up changes.